kotlin/vci-client/src/main/java/io/mosip/vciclient/authorizationServer/AuthorizationServerDiscoveryService.kt

```
@@ -16,42 +17,50 @@ class AuthorizationServerDiscoveryService {
+        for (wellKnownUrl in buildCandidateWellKnownUrls(baseUrl)) {
+            try {
+                val response = NetworkManager.sendRequest(
+                    url = wellKnownUrl,
+                    method = HttpMethod.GET,
+                    timeoutMillis = Constants.DEFAULT_NETWORK_TIMEOUT_IN_MILLIS
+                )
+                if (response.body.isNotBlank()) {
+                    JsonUtils.deserialize(response.body, AuthorizationServerMetadata::class.java)
+                        ?.let { return@withContext it }
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)